src/searchengine/searchsortmodel.cpp

```diff
@@ -0,0 +1,165 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ *
+ */
+
+#include "searchsortmodel.h"
+
+SearchSortModel::SearchSortModel(QObject *parent)
+    : base(parent)
+    , m_isNameFilterEnabled(false)
+    , m_minSeeds(-1)
+    , m_maxSeeds(-1)
+    , m_minLeechs(-1)
+    , m_maxLeechs(-1)
+    , m_minSize(-1)
+    , m_maxSize(-1)
+{
+}
+
+void SearchSortModel::enableNameFilter(bool enable)
+{
+    m_isNameFilterEnabled = enable;
+}
+
+void SearchSortModel::setNameFilter(const QString &searchTerm)
+{
+    m_searchTerm = searchTerm;
+    if (searchTerm.length() > 2 &&
+            searchTerm.at(0) == QChar('\"') && searchTerm.at(searchTerm.length() - 1) == QChar('\"')) {
+        m_searchTermWords.append(m_searchTerm.mid(1, m_searchTerm.length() - 2));
+    }
+    else {
+        m_searchTermWords = searchTerm.split(QChar(' '), QString::SkipEmptyParts);
+    }
+}
+
+void SearchSortModel::setSizeFilter(qint64 minSize, qint64 maxSize)
+{
+    m_minSize = std::max(static_cast<qint64>(-1), minSize);
+    m_maxSize = std::max(static_cast<qint64>(-1), maxSize);
+}
+
+void SearchSortModel::setSeedsFilter(int minSeeds, int maxSeeds)
+{
+    m_minSeeds = std::max(-1, minSeeds);
+    m_maxSeeds = std::max(-1, maxSeeds);
+}
+
+void SearchSortModel::setLeechsFilter(int minLeechs, int maxLeechs)
+{
+    m_minLeechs = std::max(-1, minLeechs);
+    m_maxLeechs = std::max(-1, maxLeechs);
+}
+
+bool SearchSortModel::isNameFilterEnabled() const
+{
+    return m_isNameFilterEnabled;
+}
+
+QString SearchSortModel::searchTerm() const
+{
+    return m_searchTerm;
+}
+
+int SearchSortModel::minSeeds() const
+{
+    return m_minSeeds;
+}
+
+int SearchSortModel::maxSeeds() const
+{
+    return m_maxSeeds;
+}
+
+qint64 SearchSortModel::minSize() const
+{
+    return m_minSize;
+}
+
+qint64 SearchSortModel::maxSize() const
+{
+    return m_maxSize;
+}
+
+bool SearchSortModel::lessThan(const QModelIndex &left, const QModelIndex &right) const
```

Comment:
I've opened another PR which is related to this function, if possible please use the new code:
https://github.com/qbittorrent/qBittorrent/pull/3908/files#diff-aed1f63858bd7b8e14180e4bebde4a30